platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/FileSigningUtils.java

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2021-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.util;
+
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.computeEntireHash;
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.computeMetaHash;
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.readFirstIntFromFile;
+import static com.swirlds.common.stream.internal.TimestampStreamFileWriter.writeSignatureFile;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.crypto.SignatureType;
+import com.swirlds.common.internal.SettingsCommon;
+import com.swirlds.common.stream.StreamType;
+import com.swirlds.common.stream.internal.InvalidStreamFileException;
+import com.swirlds.common.utility.ByteUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * This is a utility class to generate signature files
+ * <p>
+ * It can be used to sign v5 stream files in particular, or generally to sign any arbitrary file
+ */
+public final class FileSigningUtils {
+    /**
+     * Hidden constructor for utility class
+     */
+    private FileSigningUtils() {}
+
+    /**
+     * The suffix to append to file name when generating corresponding signature file
+     */
+    public static final String SIGNATURE_FILE_NAME_SUFFIX = "_sig";
```

Comment:
The signature extension for event files is already defined in `EventStreamType`

Reply:
`EVENT_SIG_EXTENSION = "evts_sig"` is defined, which is the full extension. However, this class needs just the suffix `_sig`, which gets appended to the original file name